includes/class-cron-options-cpt.php

```diff
@@ -0,0 +1,375 @@
+					'post_type'        => self::POST_TYPE,
+					'post_status'      => self::POST_STATUS,
+					'suppress_filters' => false,
+					'posts_per_page'   => 1,
+				) );
+
+				if ( ! $job_exists ) {
+					// Build minimum information needed to create a post
+					$job_post = array(
+						'post_title'            => $this->event_title( $event['timestamp'], $event['action'], $event['instance'] ),
+						'post_name'             => $this->event_name( $event['timestamp'], $event['action'], $event['instance'] ),
+						'post_content_filtered' => maybe_serialize( array(
+							'action'   => $event['action'],
+							'instance' => $event['instance'],
+							'args'     => $event['args'],
+						) ),
+						'post_date'             => date( 'Y-m-d H:i:s', $event['timestamp'] ),
+						'post_date_gmt'         => date( 'Y-m-d H:i:s', $event['timestamp'] ),
+						'post_type'             => self::POST_TYPE,
+						'post_status'           => self::POST_STATUS,
+					);
+
+					$this->create_job( $job_post );
+				}
+			}
+		}
+	}
+
+	/**
+	 * PLUGIN UTILITY METHODS
+	 */
+
+	/**
+	 * Retrieve list of jobs, respecting whether or not the CPT is registered
+	 */
+	private function get_jobs( $args ) {
+		// If called before `init`, we need to query directly because post types aren't registered earlier
+		if ( did_action( 'init' ) ) {
+			return get_posts( $args );
+		} else {
+			global $wpdb;
+
+			$orderby = 'date' === $args['orderby'] ? 'post_date' : $args['orderby'];
+
+			if ( isset( $args['paged'] ) ) {
+				$paged  = max( 0, $args['paged'] - 1 );
+				$offset = $paged * $args['posts_per_page'];
+			} else {
+				$offset = 0;
+			}
+
+			return $wpdb->get_results( $wpdb->prepare( "SELECT * FROM {$wpdb->posts} WHERE post_type = %s AND post_status = %s ORDER BY %s %s LIMIT %d,%d;", $args['post_type'], $args['post_status'], $orderby, $args['order'], $offset, $args['posts_per_page'] ), 'OBJECT' );
+		}
+	}
+
+	/**
+	 * Check if a job post exists, respecting Core's loading order
+	 */
+	private function job_exists( $job_post ) {
+		// If called before `init`, we need to insert directly because post types aren't registered earlier
+		if ( did_action( 'init' ) ) {
+			$exists = get_posts( $job_post );
+		} else {
+			global $wpdb;
+
+			$exists = $wpdb->get_col( $wpdb->prepare( "SELECT ID FROM {$wpdb->posts} WHERE post_name = %s AND post_type = %s AND post_status = %s LIMIT 1;", $job_post['name'], self::POST_TYPE, self::POST_STATUS ) );
+		}
+
+		return empty( $exists ) ? false : array_shift( $exists );
+	}
+
+	/**
+	 * Create a job post, respecting whether or not Core is ready for CPTs
+	 */
+	private function create_job( $job_post ) {
+		// Limit how many events to insert at once
+		if ( ! Lock::check_lock( self::LOCK, 5 ) ) {
+			return false;
+		}
+
+		// If called before `init`, we need to insert directly because post types aren't registered earlier
+		if ( did_action( 'init' ) ) {
```

Comment:
Since it calls `post_type_supports()`, it should wait until after `init`: https://core.trac.wordpress.org/browser/tags/4.6.1/src/wp-includes/post.php#L2939. That said, `post_type_supports()` doesn't error when called early, and the places it's used are ones we either don't care about (thumbnails) or ones we can overcome with filters (`wp_insert_post_empty_content`). I'll just call it directly.